ts/Extensions/Breadcrumbs.ts

```diff
@@ -0,0 +1,898 @@
+                 *
+                 * @type {Highcharts.AlignValue}
+                 */
+                align: 'left',
+
+                /**
+                 * Vertical alignment of the breadcrumbs buttons.
+                 *
+                 * @type {Highcharts.VerticalAlignValue}
+                 */
+                verticalAlign: 'top',
+
+                /**
+                 * The X offset of the breadcrumbs button group.
+                 */
+                x: 0,
+
+                /**
+                 * The Y offset of the breadcrumbs button group.
+                 */
+                y: 0
+            },
+
+            /**
+            * A CSS styles for all breadcrumbs.
+            *
+            * @type      {string}
+            * @since     next
+            */
+            style: {
+                'color': '#666666',
+                'position': 'absolute'
+            },
+
+            /**
+             * A predefined shape or symbol for the separator. Other possible
+             *  values are `'circle'`, `'square'`,`'diamond'`.
+             * Custom callbacks for symbol path generation can also be added to
+             * `Highcharts.SVGRenderer.prototype.symbols`. The callback is then
+             * used by its method name.
+             *
+             * Additionally, the URL to a graphic can be given on this form:
+             * `'url(graphic.png)'`. Note that for the image to be applied to
+             * exported charts, its URL needs to be accessible by the export
+             * server. Recommended size of image is 10x10px.
+             *
+             * @type      {string}
+             * @since     next
+             */
+            separator: 'triangle',
+
+            /**
+            * A useHTML property.
+            *
+            * @type      {boolean}
+            * @since     next
+            */
+            useHTML: false,
+
+            /**
+            * The zIndex of the group.
+            *
+            * @type      {boolean}
+            * @since     next
+            */
+            zIndex: 7
+        }
+    }
+);
+
+/**
+ * The Breadcrumbs class
+ *
+ * @private
+ * @class
+ * @name Highcharts.Breadcrumbs
+ *
+ * @param {Highcharts.Chart} chart
+ *        Chart object
+ * @param {Highcharts.Options} userOptions
+ *        User options
+ */
+class Breadcrumbs {
+    public chart: Chart = void 0 as any;
+    public breadcrumbsGroup: Highcharts.SVGElement = void 0 as any;
+    public breadcrumbsList: Array<Array<any>> = [];
+    public isTreemap: boolean = void 0 as any;
+    public level: number = -1;
+    public options: Highcharts.BreadcrumbsOptions = void 0 as any;
+
+    public constructor(chart: Chart, userOptions: Highcharts.Options) {
+        this.init(chart, userOptions);
+    }
+
+    public init(chart: Chart, userOptions: Highcharts.Options): void {
+        const chartOptions = merge(userOptions, this.options),
+            breadcrumbsOptions = chart.options.breadcrumbs;
```

Comment:
`chartOptions` seems to not used here. Can be removed?

Reply:
In the `update` method we are merging the `options` with the `userOptions` so it isn't necessary to use them here.